pkg/tsdb/cloudwatch/models/request_parser.go

```diff
@@ -198,68 +215,16 @@ func parseRequestQuery(dataQuery metricsDataQuery, refId string, startTime time.
 			}
 		}
 	} else {
-		if reNumber.Match([]byte(p)) {
-			period, err = strconv.Atoi(p)
-			if err != nil {
-				return nil, fmt.Errorf("failed to parse period as integer: %v", err)
-			}
-		} else {
-			d, err := time.ParseDuration(p)
+		period, err = strconv.Atoi(periodString)
+		if err != nil {
+			d, err := time.ParseDuration(periodString)
 			if err != nil {
-				return nil, fmt.Errorf("failed to parse period as duration: %v", err)
+				return 0, fmt.Errorf("failed to parse period as duration: %v", err)
```

Comment:
Most Go code I've seen usually returns `nil` if there is an error instead of a value. Is there a reason for returning `0` here instead of `nil`?

Reply:
Yup, I can't actually return `nil` here because of the `int` type. 
The convention is to return the "zero value" alongside an error. Often the zero values are nil, so that's why you've seen that often.

Here's an article that has types and their zero values https://nilsmagnus.github.io/post/nillability-in-go/
TLDR: This is what I remember off the top of my head and then when needed I look up others: Basic types are not nullable. Structs aren't either. Slices and maps are. (Maps are a common "gotcha" -- they need to be initialized, for example ` var fluffy = make(map[string]string)` or `var fluffy = map[string]string{}` before you start inserting into them. Often we forget to do this.)